packages/firestore/index.node.ts

```diff
@@ -19,6 +19,7 @@ import { FirebaseNamespace } from '@firebase/app-types';
 
 import { Firestore, IndexedDbPersistenceProvider } from './src/api/database';
 import { configureForFirebase } from './src/config';
+import { FirebaseFirestore as ExpFirebaseFirestore } from './exp/src/api/database';
```

Comment:
I think we actually need to rename the one renaming Firestore class (the legacy class) to FirebaseFirestore as well. The problem is that we want to use the class name in the error messages (see https://github.com/firebase/firebase-js-sdk/blob/master/packages/firestore/src/util/input_validation.ts#L179). If we rename the classes in the source, the new name leaks via the `constructor.name` property and the check will no longer print the more helpful error message.

So I think the way forward should be:
- Rename the Firestore class in api/database.ts to FirebaseFirestore.
- Settle on a prefix or suffix notation.

Reply:
I will send a follow-up PR that addresses these issues shortly.

Reply:
Renaming the internals breaks the minifier since the renamed symbols are not considered part of the public API. This doesn't affect the "Public" rename, so I will still go ahead with that.